img2dataset/downloader.py

```diff
@@ -17,6 +17,27 @@
 from .logger import write_stats
 
 
+DISALLOWED_DIRECTIVES = ("noindex", "noimageindex", "noai", "noimageai")
+USER_AGENT_TOKEN = "img2dataset"
+
+
+def is_disallowed(headers):
```

Comment:
This does one additional call, so can you check speed ?

I am not convinced this should be done at the downloading step but rather at the crawling step (while collecting the links)

Reply:
Note there is no change to HTTP requests being issued -- the headers are already sent (or not) with the file download. Are you concerned with overhead associated with the additional python method call to parse headers? I will include timings and stats for this branch compared to current main head branch in a bit (running now) but i suspect these overheads will be negligible compared to the cost of the actual HTTP requests.

Re your other point, I agree that the crawling step should check page HTML for relevant robots meta tag directives and omit images accordingly.  If there is a repo used for indexing the LAION datasets that you'd suggest adding this functionality to, I'd be happy to work on a PR for that as well.   

However, it's possible that the creator of a given dataset did NOT consider such directives when their crawl was done, and it's also possible that directives associated with a specific media file change after the crawl (the media owner has every right to do so). As such I don't think the user who actually downloads and uses the indexed media can be absolved of all responsibility to respect directives that apply to their intended usage, especially if those directives are made readily available. Since the HTTP headers are sent with the image files when downloaded anyway, I worry it could be seen as negligent by the img2dataset user to receive these directive but choose to ignore them -- especially now that legality of use of third-party images for AI training purposes is being tested.

Reply:
ok, let's add this under a command line option